mbed/mbed.py

```diff
@@ -642,9 +658,12 @@ def publish(all_refs=None):
                 if not branch:
                     error(err+"Working set is not on a branch.", 1)
 
-    def fetch():
+    def fetch(url=None, rev=None, depth=None):
         info("Fetching revisions from remote repository to \"%s\"" % os.path.basename(os.getcwd()))
-        popen([git_cmd, 'fetch', '--all', '--tags'] + (['-v'] if very_verbose else ([] if verbose else ['-q'])))
+        if url:
+            popen([git_cmd, 'fetch', '--tags'] + ([url] if url else []) + ([rev] if rev else []) + (["--depth", depth] if depth else []) + (['-v'] if very_verbose else ([] if verbose else ['-q'])))
```

Comment:
URL should be passed to format(url, protocol) for behavior consistency. See line 627 `popen([git_cmd, 'clone', formaturl(url, protocol), path] + (['-v'] if very_verbose else ([] if verbose else ['-q'])))`